Source/EasyNetQ/Producer/PublishConfirmationListener.cs

```diff
@@ -0,0 +1,126 @@
+using System;
+using System.Collections.Concurrent;
+using System.Linq;
+using System.Threading;
+using System.Threading.Tasks;
+using EasyNetQ.Events;
+
+namespace EasyNetQ.Producer
+{
+    public class PublishConfirmationListener : IPublishConfirmationListener
+    {
+        private ConcurrentDictionary<ulong, TaskCompletionSource<NullStruct>> unconfirmedRequests = new ConcurrentDictionary<ulong, TaskCompletionSource<NullStruct>>();
+        private readonly CancellationTokenSource cancellation = new CancellationTokenSource();
+
+        public PublishConfirmationListener(IEventBus eventBus)
+        {
+            eventBus.Subscribe<MessageConfirmationEvent>(OnMessageConfirmation);
+            eventBus.Subscribe<PublishChannelCreatedEvent>(OnPublishChannelCreated);
+        }
+
+        public void Request(ulong deliveryTag)
+        {
+            unconfirmedRequests.Add(deliveryTag, new TaskCompletionSource<NullStruct>());
+        }
+
+        public void Discard(ulong deliveryTag)
+        {
+            unconfirmedRequests.Remove(deliveryTag);
+        }
+
+        public void Wait(ulong sequenceNumber, TimeSpan timeout)
```

Comment:
use `deliveryTag` instead of `sequenceNumber` to be the same as [Request](https://github.com/EasyNetQ/EasyNetQ/pull/461/files#diff-d27d22999c24786e76dc0cc98b994a44R21) and [Discard](https://github.com/EasyNetQ/EasyNetQ/pull/461/files#diff-d27d22999c24786e76dc0cc98b994a44R26) method.